bpdm-pool/src/main/kotlin/org/eclipse/tractusx/bpdm/pool/service/BusinessPartnerChangesDetector.kt

```diff
@@ -0,0 +1,259 @@
+/*******************************************************************************
+ * Copyright (c) 2021,2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package org.eclipse.tractusx.bpdm.pool.service
+
+import org.eclipse.tractusx.bpdm.pool.entity.*
+import org.springframework.stereotype.Service
+
+@Service
+class BusinessPartnerChangesDetector {
+    fun haveLegalEntityChanges(original: LegalEntity, updated: LegalEntity): Boolean {
+        return (
+                original.bpn != updated.bpn ||
+                        haveLegalFormChanges(original.legalForm, updated.legalForm) ||
+                        haveSetIdentifiersChanges(original.identifiers, updated.identifiers) ||
+                        haveSetStateChanges(original.states, updated.states) ||
+                        haveSetLogisticAddressChanges(original.addresses, updated.addresses) ||
+                        haveSetSiteChanges(original.sites, updated.sites) ||
+                        haveSetClassificationChanges(original.classifications, updated.classifications) ||
+                        haveLegalNameChanges(original.legalName, updated.legalName) ||
+                        haveLogisticAddressChanges(original.legalAddress, updated.legalAddress)
+                )
+    }
+
+
+    fun haveSetIdentifiersChanges(originalSet: MutableSet<LegalEntityIdentifier>, updatedSet: MutableSet<LegalEntityIdentifier>): Boolean {
```

Comment:
Instead of having different methods for each existing set, maybe create a generic method comparing two lists which accepts a comparator function. You would then use that function every time you want to compare sets or lists with each other. Have a look here for inspiration: https://stackoverflow.com/questions/52054104/comparing-two-lists-in-kotlin